middleware/middleware_session.go

```diff
@@ -0,0 +1,129 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/0xsequence/quotacontrol/proto"
+)
+
+// Session middleware that detects the session type and sets the account or service on the context.
+func Session() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return session{Next: next}
+	}
+}
+
+// AccessControl middleware that checks if the session type is allowed to access the endpoint.
+// It also sets the compute units on the context if the endpoint requires it.
+func AccessControl(acl ACL, cost Cost, defaultCost int64) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return accessControl{
+			ACL:         acl,
+			Cost:        cost,
+			DefaultCost: defaultCost,
+			Next:        next,
+		}
+	}
+}
+
+type session struct {
+	Next http.Handler
+}
```

Comment:
I think this is unnecessary abstraction

Reply:
See https://github.com/0xsequence/quotacontrol/pull/41#discussion_r1664128802